### PR TITLE
feat: switch dep to go module

### LINKED ISSUE #20: Switch glide to go modules for dependency management
**Is your feature request related to a problem?/Why is this needed**

**Describe the solution you'd like in detail**

could refer to https://github.com/kubernetes/cloud-provider-azure/pull/142

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by ZeroMagic**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Switch dep to go module.
Remove snapshot related code in e2e framework, because it is unnecessary now.
Need to use go-toml less than v1.8, otherwise there will be an error when parsing the toml file.

**Which issue(s) this PR fixes**:

Fixes #20 

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:
Refer to https://github.com/kubernetes-sigs/azurefile-csi-driver/pull/236

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/util/util.go [hunk 1]
```diff
@@ -66,3 +70,13 @@ func GetMountOptions(options []string) string {
 	}
 	return str
 }
+
+func MakeDir(pathname string) error {
```
Comment (andyzhangx) [inline pkg/util/util.go:74]: could you add a unit for this function @Sakuralbj

File: hack/update-gomod.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
```
Comment (andyzhangx) [inline hack/update-gomod.sh:1]: shall we always use this tool when update-gomod?

Comment (ZeroMagic) [reply hack/update-gomod.sh:1]: No. Just use it to upgrade Kubernetes.

File: pkg/util/util.go (hunk 1 above)
Comment (Sakuralbj) [reply pkg/util/util.go:74]: ok

**Pull request merged**